plugins/catalog-backend-module-msgraph/src/service/MicrosoftGraphOrgEntityProviderCatalogModule.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  createBackendModule,
+  loggerToWinstonLogger,
+  configServiceRef,
+  loggerServiceRef,
+  schedulerServiceRef,
+} from '@backstage/backend-plugin-api';
+import { catalogProcessingExtensionPoint } from '@backstage/plugin-catalog-node';
+import { MicrosoftGraphOrgEntityProvider } from '../processors';
+
+/**
+ * Registers the MicrosoftGraphOrgEntityProvider with the catalog processing extension point.
+ *
+ * @alpha
+ */
+export const microsoftGraphOrgEntityProviderCatalogModule = createBackendModule(
+  {
+    pluginId: 'catalog',
+    moduleId: 'microsoftGraphOrgEntityProvider',
+    register(env) {
+      env.registerInit({
+        deps: {
+          config: configServiceRef,
+          catalog: catalogProcessingExtensionPoint,
+          logger: loggerServiceRef,
+          scheduler: schedulerServiceRef,
+        },
+        async init({ config, catalog, logger, scheduler }) {
+          catalog.addEntityProvider(
+            MicrosoftGraphOrgEntityProvider.fromConfig(config, {
+              logger: loggerToWinstonLogger(logger),
+              scheduler,
```

Comment:
here, we should support the custom transformations. Maybe via extension point or option

Reply:
I realized extension points are not supported at modules. Hence, I will go for options for now. Maybe extension points can be better though 🤔

Reply:
see also discussion at https://github.com/backstage/backstage/pull/14039#discussion_r992442514